clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -324,62 +327,6 @@ private long listOffset(TopicPartition partition, long timestamp) {
     }
 
     /**
-     * If any partition from previous fetchResponse contains OffsetOutOfRange error and
-     * the defaultResetPolicy is NONE, throw OffsetOutOfRangeException
-     *
-     * @throws OffsetOutOfRangeException If there is OffsetOutOfRange error in fetchResponse
-     */
-    private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
```

Comment:
Yes, that is correct. I didn't see a clear advantage to bundling them into a single exception, especially not one that outweighed the benefit of being able to parse separately. As far as a test case, I don't know... It doesn't seem to add a lot of value since the fetch grouping is really an implementation detail which doesn't affect usage semantics.